gateway/gateway_test.go

```diff
@@ -1115,3 +1238,16 @@ func getWorkspaceID(writeKey string) string {
 	defer configSubscriberLock.RUnlock()
 	return enabledWriteKeyWorkspaceMap[writeKey]
 }
+
+type mockRequestHandler struct{}
+
+func (mockRequestHandler) ProcessRequest(gateway *HandleT, w *http.ResponseWriter, r *http.Request, reqType string, payload []byte, writeKey string) string {
+	// deepsource ignore: Unused method arguments
+	_ = gateway
+	_ = w
+	_ = r
+	_ = reqType
+	_ = payload
+	_ = writeKey
+	return ""
```

Comment:
btw for deepsource you can do this as well:
```suggestion
func (mockRequestHandler) ProcessRequest(_ *HandleT, _ *http.ResponseWriter, _ *http.Request, _ string, _ []byte, _ string) string {
```